Jenkinsfile

```diff
@@ -86,6 +86,30 @@ pipeline {
         }
       }
     }
+    stage('Lint') {
+      options { skipDefaultCheckout() }
+      environment {
+        PATH = "${env.PATH}:${env.WORKSPACE}/bin"
+        HOME = "${env.WORKSPACE}"
+        GOPATH = "${env.WORKSPACE}"
+      }
+      when {
+        beforeAgent true
+        allOf {
+          expression { return params.intake_ci }
+          expression { return env.ONLY_DOCS == "false" }
+        }
+      }
+      steps {
+        withGithubNotify(context: 'Lint') {
+          deleteDir()
+          unstash 'source'
+          dir("${BASE_DIR}"){
+            sh(label: 'Run Lint', script: './.ci/scripts/lint.sh')
```

Comment:
NIT:

What do you think a make goal with the script content? Something like the below snippet?

```
ci-lint: lint
  git status --porcelain || (echo "Lint detected files not following project's format. Please run 'make fmt' locally to fix this error"; exit 1) && echo 'Lint passed'
```

Reply:
Sent in 5fd4f3d